pkg/api/getambassador.io/v1/crd_devportal.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2020 Datawire.  All rights reserved
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	ambv2 "github.com/datawire/ambassador/v2/pkg/api/getambassador.io/v2"
+)
+
+// DevPortal is the Schema for the DevPortals API
+//
+// DevPortal resources specify the `what` and `how` is shown in a DevPortal:
+//
+// * `what` is in a DevPortal can be controlled with
+//   - a `selector`, that can be used for filtering `Mappings`.
+//   - a `docs` listing of (services, url)
+// * `how` is a pointer to some `contents` (a checkout of a Git repository
+//   with go-templates/markdown/css).
+//
+// Multiple `DevPortal`s can exist in the cluster, and the Dev Portal server
+// will show them at different endpoints. A `DevPortal` resource with a special
+// name, `ambassador`, will be used for configuring the default Dev Portal
+// (served at `/docs/` by default).
+//
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=devportals,scope=Namespaced
+type DevPortal struct {
```

Comment:
To put it somewhere public: from the private #emissary-maintainers channel: 

@kflynn:
> One of @LanceEa’s comments on #4055 has me wondering about `DevPortal` – do I remember correctly that it’s in the Emissary CRDs because way TF back when we didn’t have separate Edge Stack CRDs? Presumably we could move it now, right?

@LanceEA:
> Yeah, I haven’t played with this feature much but I had thought they were an EdgeStack only feature so what your saying makes sense to me if that is true.
>
> Does any of the generated routes in diagd look at these? If so then that is probably part of the isEdgeStack things that we need to address before we could move these.

@kflynn:
> AFAIK there is *literally* nothing in Emissary that understands `DevPortal` at all.
>
> This is definitely a place where I’d love to get confirmation from @AliceProxy and @LukeShu though.

@AliceProxy:
> That's also what I thought

@LukeShu:
> We *always* had a way for them to be separate.
>
> So I'm sure there has to be a reason it went in Emissary.

@kflynn:
> Searching for `DevPortal` in the Emissary directory, all I see are tests and a comment in `pkg/agent/api_docs.go` about “be careful because there’s a similar scraping implementation in DevPortal”.
>
> `DevPortals` do appear in the snapshot; is that so Cloud can see them? are the Edge Stack types not sent back to Cloud?

@AliceProxy:
> Probably for the cloud devportal stuff
>
> As far as I'm aware, anything in the snapshot should be sent to the cloud.

@kflynn:
> Well that’s too bad. :joy:

@LukeShu:
> The PR to blame: https://github.com/datawire/apro/pull/2007
>
> I think the answer is "Alvaro didn't know how to add AES-only CRDs, and Flynn didn't catch it in code review"
>
> No! I think the answer is "So that it could take advantage of the main watcher/snapshot, instead of having to implement another watcher in amb-sidecar"

@AliceProxy:
> That was my reason for moving Filters over to Interesting Types in emissary
>
> (And so you could process the secret refs in the reconciler steps)

@kflynn:
> I think #2 is the reason, yeah. Damn. I’d forgotten about that.